### PR TITLE
slack connector: handle thread not found

### PR DESCRIPTION
**Pull request by spolu**
Fixes: https://github.com/dust-tt/tasks/issues/130

### CODE REVIEW

File: connectors/src/connectors/slack/temporal/activities.ts [hunk 1]
```diff
@@ -493,33 +494,51 @@ export async function syncThread(
...
+      }
+      allMessages = allMessages.concat(
+        replies.messages.filter((m) => !!m.user)
+      );
+      next_cursor = replies.response_metadata?.next_cursor;
+    } catch (e) {
+      const slackError = e as CodedError;
+      if (slackError.code === ErrorCode.PlatformError) {
+        const platformError = slackError as WebAPIPlatformError;
+        if (platformError.data.error === "channel_not_found") {
```
Comment (lasryaric) [inline connectors/src/connectors/slack/temporal/activities.ts:529]: wasn't the error `thread_not_found` instead of `channel_not_found` ?

Comment (spolu) [reply connectors/src/connectors/slack/temporal/activities.ts:529]: ah yeah fixing

Comment (spolu) [reply connectors/src/connectors/slack/temporal/activities.ts:529]: thanks for catching

**Review by lasryaric — CHANGES_REQUESTED** [review]
Need change.

File: connectors/src/connectors/slack/temporal/activities.ts [hunk 2]
```diff
@@ -493,33 +494,51 @@ export async function syncThread(
...
+      allMessages = allMessages.concat(
+        replies.messages.filter((m) => !!m.user)
+      );
+      next_cursor = replies.response_metadata?.next_cursor;
+    } catch (e) {
+      const slackError = e as CodedError;
+      if (slackError.code === ErrorCode.PlatformError) {
+        const platformError = slackError as WebAPIPlatformError;
+        if (platformError.data.error === "thread_not_found") {
+          break;
```
Comment (lasryaric) [inline connectors/src/connectors/slack/temporal/activities.ts:530]: I would return here directly. There is no point in keeping going.

**Review by lasryaric — APPROVED** [review]

**Pull request merged**